kb/migrating-visibility-data-from-es6.md

```diff
@@ -0,0 +1,74 @@
+---
+slug: migrate-visibility-data-from-es6
+title: Migrate visibility data from ES6
+tags:
+  - kb-article
+date: 2022-29-08T00:00:00Z
+---
+
+We added support for [Elasticsearch v7+ (ES7)](https://www.elastic.co/downloads/past-releases/#elasticsearch) in the v1.7.0 update to the [Temporal Server](/cluster/s#temporal-server).
+Elasticsearch v7 introduces several breaking changes, including the removal of mapping types.
+These changes make Elasticsearch v6 incompatible with Elasticsearch v7.
+
+As of v1.18.0, v6 support has been completely removed from Temporal.
+Support for Elasticsearch v8 has been added; v7 remains as the default version to run.
+
+Use one of the following methods to update your Temporal Server and Elasticsearch.
+
+## Temporary shutdown
+
+1. Shut down the Temporal Cluster.
+2. Upgrade Elasticsearch v6 to Elasticsearch v7 according to [Elasticsearch documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/setup-upgrade.html).
+3. Upgrade Temporal to the latest version.
+4. Make sure that the static config file is set to `v7`:
+
+   ```
+   persistence:
+     datastores:
+       es-visibility:
+         elasticsearch:
+           version: "v7"
+   ```
+
+5. Start the Temporal Cluster.
+
+## Rolling upgrade
+
+1. If you are still using Elasticsearch v6, make sure your config file is set to `v6`:
+   ```
+   persistence:
+     datastores:
+       es-visibility:
+         elasticsearch:
+           version: "v6"
+   ```
+   If you're using a pre-build docker image, set `ES_VERSION` to `v6`.
+2. Update Temporal to v1.17.5.
```

Comment:
I would insist on my original statement here. Who knows how many patches we will create for 1.17. I am preparing 1.17.6 right now. Or you can say "Update Temporal to the latest 1.17.x" version.